### DISCUSSION TITLE
Trouble with input on VM with NVidia card

### DISCUSSION THREAD

**Discussion by TheJenious0917** [LizardByte/community#14] [Sunshine] [answered]
Hey ya'll, want to see if maybe someone has run across this and maybe I'm just crazy or if this is some sort of bug. I'm currently running a POP_OS 21.10 VM on ProxMox, with a NVidia card attached via pcie passthrough. In proxmox if I configure the VM with a "default" software display, Sunshine will stream just fine and input works as expected. No issues, other than I'm just not utilizing the GPU. 

However, once I remove the default display and switch to the GPU as the primary display, the VM still works, driver is working as intended (as far as I can tell), RDP to the machine works normally, but when I connect to it via Sunshine, input no longer works. I've reviewed the sunshine logs and there's nothing about permission issues, or errors at all that I can find. I've run across the mouse permission denied thing before and solved with the udev rules permission change fix. 

I know that input is detected at least partially, as I have Sunshine set to launch retroarch and it launches, but I get a RetroArch notification that says the connected Xbox controller is not configured. I know it sees that it exists, it just doesn't react to it, or mouse and keyboard either. 

Anybody know of fixes for this, or at least point me towards some things to check for errors or something?
Thanks

**Comment by TheJenious0917** [comment]
For a little more detail, I do have a dummy display adapter plug in the GPU. When connecting via RDP to the VM, i can launch sunshine and this is the log around the time I connect using Moonlight. I guess I do see something that says a warning about a sink, but not sure if that has anything to do with the input issue. I also noticed that it says I'm streaming display rdp0, so I'm wondering if something about being logged in first via an RDP session is causing some conflict? Strangely, when i echo $DISPLAY it says :10.0 is my display, not rdp0, but I may be misunderstanding something there. 

```txt
[2022:02:25:16:48:37]: Info: /CN=NVIDIA GameStream Client -- verfied
[2022:02:25:16:48:37]: Info: CLIENT CONNECTED
[2022:02:25:16:48:37]: Info: Detecting connected monitors
[2022:02:25:16:48:37]: Info: Screencasting with X11
[2022:02:25:16:48:37]: Info: Configuring selected monitor (0) to stream
[2022:02:25:16:48:37]: Info: Streaming display: rdp0 with res 1920x1080 offset by 0x0
[2022:02:25:16:48:37]: Info: Color coding [Rec. 709]
[2022:02:25:16:48:37]: Info: Color range: [MPEG]
[2022:02:25:16:48:38]: Warning: Couldn't find an active sink
[2022:02:25:16:48:58]: Info: CLIENT DISCONNECTED
[2022:02:25:16:48:58]: Info: /CN=NVIDIA GameStream Client -- verfied
```

**Comment by TheJenious0917** [comment, marked as answer]
Hey, figured I'd update this. It is now working and input is being detected properly. I think it was potentially an Nvidia driver issue. I upgraded the drivers up to 470, then 510 just to see if it performed differently. At 510, I happened to check dmesg and noticed that there was an error that said my card was only supported up to 470 drivers. I rolled back and miraculously it worked (despite already having tried that version). Anyway, it's now working and I think this can be closed.

**Reply by ReenigneArcher** [reply]
Thanks for the update!
